### PR TITLE
Update deploy-vue-storefront-cloud.yml

### PR DESCRIPTION
**Pull request by renanoliveira0**
https://vsf.atlassian.net/browse/PL-1465

### CODE REVIEW

**Review by Baroshem — APPROVED** [review]

**Review by FRSgit — APPROVED** [review]

**Review by MarcinSulowski — APPROVED** [review]

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (nestak) [inline .github/workflows/deploy-vue-storefront-cloud.yml]: issue: Why it is the `CLOUD_API_BASE_URL` variable? It will point to Farmer API. I thought you said all requests should go through platform api?

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (nestak) [inline .github/workflows/deploy-vue-storefront-cloud.yml]: issue: If we go through Platform's api with deployments, then both requests can be hidden, and made by Platform to Farmer.

**Review by nestak — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (radzikowski) [inline .github/workflows/deploy-vue-storefront-cloud.yml]: Issue: this should be "deploy" because our cloud.service ciTemplate will crash...

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (renanoliveira0) [reply .github/workflows/deploy-vue-storefront-cloud.yml]: Excellent observation, you are completely correct.

Done: a94efad32eab750005e153d3bc954223758603aa

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (renanoliveira0) [reply .github/workflows/deploy-vue-storefront-cloud.yml]: Done: a94efad32eab750005e153d3bc954223758603aa

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (renanoliveira0) [reply .github/workflows/deploy-vue-storefront-cloud.yml]: Initially I thought it was an excellent idea, but on second thought, I don't think that's the case.

If we only make a single request here in the Github Actions CI/CD script and leave the two requests on the Platform/Console side, we will deliver a bad user experience, as a single error may occur in a single deploy of the two and the user will not know exactly what the problem is, as he will only receive an error from Github Actions stating that there was an error. By keeping requests separate, we give more information to the user who can get straight to the issue at hand.

**Review by Baroshem — APPROVED** [review]

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (radzikowski) [inline .github/workflows/deploy-vue-storefront-cloud.yml:70]: Issue: Why do we have to all the time define path and port in every deployment?

**Review by radzikowski — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (nestak) [reply .github/workflows/deploy-vue-storefront-cloud.yml]: Good point.

**Review by nestak — APPROVED** [review]

File: .github/workflows/deploy-vue-storefront-cloud.yml
Comment (renanoliveira0) [reply .github/workflows/deploy-vue-storefront-cloud.yml:70]: https://github.com/vuestorefront/platform-backend/pull/340#discussion_r1043098433

https://github.com/vuestorefront/platform-backend/pull/340#discussion_r1053751614

Comment (radzikowski) [reply .github/workflows/deploy-vue-storefront-cloud.yml:70]: But if you look at this, every time on every github action trigger, we put some data in this request, what is the purpose? - changing these parameters after every deployment? check is the same and fail? Could you explain it to me?

Comment (renanoliveira0) [reply .github/workflows/deploy-vue-storefront-cloud.yml:70]: This was done like this by Anastazy, as you can see:

https://github.com/vuestorefront/sample-express-application-with-middleware/commits/main

Here I don't really change anything, I just remove some things that in my view are static, and I leave them in the Platform/Console and I leave here the things that can be changed.

If in fact they are changed, how often, by whom, etc., I don't know. It was done that way, and I have no guidelines or context to change it, after all, this is a very critical part of the system. I don't believe that changing something without an explicit and important reason is the right thing to do.

**Review by Baroshem — APPROVED** [review]

**Comment by renanoliveira0** [comment]
WARNING!

This PR can only be merged when this other PR [here](https://github.com/vuestorefront/platform-backend/commit/6c8f28e5b5f77f59e8b6fa3f6804dd25a5ee6e13) is in production.

**Review by RetroModernDev — CHANGES_REQUESTED** [review]
As I mentioned in Jira, 

we need to find a solution to test this on the test/stage. env 

Testing on production is not acceptable.

**Review by RetroModernDev — APPROVED** [review]

**Comment by renanoliveira0** [comment]
@Frodigo, 

The critical subject here.

This PR here was finalized, and everything is working perfectly in testing: https://vsf.atlassian.net/browse/PL-1465

However, even if this PR code goes up to production, this will not automatically imply production correction.

What will fix production is this PR here (https://github.com/vuestorefront/sample-express-application-with-middleware/pull/2), which needs to be manually merged.

However, it can only be merged after https://github.com/vuestorefront/platform-backend/commit/7f0aefa76c3358bf43c3f6e1619f0eafc065373c is in production on Platform/Console, if done before then it will break middleware deployment.

Even though everything was previously tested in a test environment, after the merge of this PR a test in production is necessary.

**Comment by renanoliveira0** [comment]
As I explained earlier, the lack of this PR here in production implies that every new repository created is being created with the wrong script and later in the future will have to be fixed by hand, one by one.

So, in my opinion, at least, uploading the PR https://github.com/vuestorefront/platform-backend/commit/7f0aefa76c3358bf43c3f6e1619f0eafc065373c to production so that we can merge this PR here is something that should be done as soon as possible, when instead of waiting for a future release.

I leave it up to you to upload this PR to production separately as soon as possible, merge this PR here, and ask Magdalena to carry out all the tests in production.

**Comment by renanoliveira0** [comment]
This PR must remain closed.

It is not possible to change it and it has an error in the deploy name.

**Pull request closed without merging**
